cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -608,15 +605,17 @@ updateCreateStakedOutputTemplate
           , sgProtocolParams
           }
   where
+    nUtxoAddrsNonDeleg  = length utxoAddrsNonDeleg
     maximumLovelaceSupply :: Word64
     maximumLovelaceSupply = sgMaxLovelaceSupply template
     -- If the initial funds are equal to the maximum funds, rewards cannot be created.
     subtractForTreasury :: Integer
     subtractForTreasury = nonDelegCoin `quot` 10
-    nonDelegCoin, delegCoin :: Integer
-    -- if --supply is not specified, non delegated supply comes from the template passed to this function:
-    nonDelegCoin = fromIntegral (maybe maximumLovelaceSupply unLovelace mAmountNonDeleg)
-    delegCoin = maybe 0 fromIntegral amountDeleg
+    totalSupply, nonDelegCoin, delegCoin :: Integer
+    -- if --total-supply is not specified, supply comes from the template passed to this function:
+    totalSupply = fromIntegral (maybe maximumLovelaceSupply unLovelace mTotalSupply)
```

Comment:
Why not modify: `sgMaxLovelaceSupply = fromIntegral $ nonDelegCoin + delegCoin`
to `sgMaxLovelaceSupply = totalSupply`

Reply:
Good catch, done :+1: